### PR TITLE
mb/system76/cml-u: Convert lemp9 to a variant

### PR DESCRIPTION
**Pull request by crawfxrd**

### CODE REVIEW

**Comment by linuxgnuru** [comment]
After building and flashing the lemp9 was bricked

**Comment by crawfxrd** [comment]
Likely due to a typo, which for some reason compiles, but breaks memory configuration.

File: src/mainboard/system76/cml-u/Kconfig [hunk 1]
```diff
@@ -11,13 +11,15 @@ config BOARD_SPECIFIC_OPTIONS
...
 	select INTEL_GMA_HAVE_VBT
 	select INTEL_LPSS_UART_FOR_CONSOLE
 	select MEMORY_MAPPED_TPM
 	select MAINBOARD_HAS_TPM2
 	select NO_UART_ON_SUPERIO
-	select PCIEXP_HOTPLUG
-	select PCIEXP_HOTPLUG_PREFETCH_MEM_BELOW_4G # Fix running out of MTRRs
+	select PCIEXP_HOTPLUG if BOARD_SYSTEM76_DARP6 || BOARD_SYSTEM76_GALP4
+	# Fix running out of MTRRs
+	select PCIEXP_HOTPLUG_PREFETCH_MEM_BELOW_4G if BOARD_SYSTEM76_DARP6 || BOARD_SYSTEM76_DARP6
```
Comment (jackpot51) [inline src/mainboard/system76/cml-u/Kconfig:22]: This lists DARP6 twice

**Comment by linuxgnuru** [comment]
When running `./scripts/build.sh lemp9` I get the following error:

```
Cleaning up temporary files... ok

You can now run i386-elf cross GCC from /home/system76/firmware-open/coreboot/util/crossgcc/xgcc.
Welcome to the coreboot cross toolchain builder v2022-04-04_9a8d0a03db 

Building toolchain using 8 thread(s).

Downloading and verifying tarballs ...
 * nasm-2.15.05.tar.bz2 (downloading from https://www.nasm.us/pub/nasm/releasebuilds/2.15.05/nasm-2.15.05.tar.bz2)...   0%... Failed to download nasm-2.15.05.tar.bz2.
make[2]: *** [Makefile:31: build_nasm] Error 1
make[1]: *** [Makefile:35: build-i386] Error 2
make: *** [util/crossgcc/Makefile.inc:30: crossgcc-i386] Error 2
```

**Comment by jacobgkau** [comment]
@linuxgnuru Those downloads sometimes time out because of the servers they're on, you may want to try again with these commands (or just starting over):

```
cd coreboot
make iasl
make crossgcc-i386
cd ..
```

**Comment by crawfxrd** [comment]
Not sure what it could be without coreboot logs. Only things I notice are `MAX_CPUS` (which is now set for lemp9) and using `mainboard_silicon_init_params()` instead of `mainboard_init()`.

**Comment by linuxgnuru** [comment]
Where would I find these logs to share?

**Comment by linuxgnuru** [comment]
```
 [11:57:26] Entering console mode
[11:57:26] 
[11:57:26] System76 EC board 'system76/lemp9', version '2022-07-07_1add7ee-dirty'
[11:57:29] Lid open
[11:57:35] 84: Power switch press
[11:57:35] 84: power_on
[11:57:35] VA_EC_EN = true
[11:57:35] DD_ON = true
[11:57:35] SUS_PWR_ACK = true
[11:57:35] PCH_DPWROK_EC = true
[11:57:35] EC_RSMRST_N = true
[11:57:35] EC_EN = true
[11:57:35] PWR_BTN_N = false
[11:57:35] PWR_BTN_N = true
[11:57:35] POWER_STATE_S0
[11:57:35] reached S0 in 0 ms
[11:57:35] 84: ALL_SYS_PWRGD asserted
[11:57:35] PM_PWROK = true
[11:57:35] PCH_PWROK_EC = true
[11:57:35] 87: Power switch press
[11:57:35] D5: Power switch release
[11:57:35] B7: PLT_RST# de-asserted
[11:57:35] Enable PNP devices
[11:57:36] sk/romstage' @8@B3 ms

[11:57:36] [DEBUG]  TPM: Digest of `CBFS: bootblock` to PCR 2 logged
[11:57:36] [DEBUG]  CRTM initialized.
[11:57:36] [DEBUG]  TPM: Digest of `CBFS: fallback/romstage` to PCR 2 logged
[11:57:36] [DEBUG]  BS: bootblock times (exec / console): total (unknown) / 33 ms
```

**Comment by crawfxrd** [comment]
The logs are corrupted, so I'm not sure what it's dying on.

**Comment by crawfxrd** [comment]
Updated for new coreboot rebase.

**Review by jackpot51 — APPROVED** [review]

**Comment by crawfxrd** [comment]
This has been upstreamed and I will deal with any remaining issues on next rebase.

**Pull request closed without merging**